MapboxMobileEvents/MMEUIApplicationWrapper.m

```diff
@@ -1,17 +1,92 @@
 #import "MMEUIApplicationWrapper.h"
+#import "NSBundle+MMEMobileEvents.h"
+#import "UIKit+MMEMobileEvents.h"
+
+// MARK: - App Extension Support
+
+/*! Provides Wrapper Getters for Non UIApplication Interfaces */
+@interface MMEUIApplicationExtensionWrapper : NSObject <MMEUIApplicationWrapper>
+@end
+
+
+@implementation MMEUIApplicationExtensionWrapper
+
+// MARK: - Properties
+
+- (NSInteger)mme_contentSizeScale {
+    return NSExtensionContext.mme_contentSizeScale;
+}
+
+- (UIApplicationState)applicationState {
+    return UIApplicationStateActive;
+}
+
+// MARK: - Methods
+
+- (UIBackgroundTaskIdentifier)beginBackgroundTaskWithExpirationHandler:(void(^ __nullable)(void))handler {
+    // No-Op. Extensions don't support background tasks
+    // Potentially leverage
+    // https://developer.apple.com/documentation/foundation/nsprocessinfo/1617030-performexpiringactivitywithreaso?language=objc
+    return UIBackgroundTaskInvalid;
+}
+
+- (void)endBackgroundTask:(UIBackgroundTaskIdentifier)identifier {
+    // no-op
+}
+
+@end
+
+@interface MMEUIApplicationWrapper ()
+@property (nonatomic, strong) id <MMEUIApplicationWrapper> application;
+@end
 
 @implementation MMEUIApplicationWrapper
 
+// MARK: - Initializers
+
+- (instancetype)init {
```

Comment:
This class serves as the adaptor layer between the framework and the ability to access data/methods from either sharedApplication, or extension friendly sources.

Some assumption have been made to prioritize ability to build over fully flushed extenion behavior 

eg.
- backgroundTasks do not function in extensions behaving as a no-op
- values are read from the extension's Info.plist

There are options to consider for further optimizations:
- Differing behaviors for extensions:
  - delayed launch?
  - Background URLSession? Allowing for tasks to be passed off to the main application to send
  - Observing NSExtension specific notifications in case we wanted to write out data somewhere which could be read later

But for now, prioritized ability to build over flushed out behavior expectations.